sycl/plugins/unified_runtime/ur/adapters/hip/enqueue.cpp

```diff
@@ -252,14 +252,23 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunch(
   std::unique_ptr<ur_event_handle_t_> RetImplEvent{nullptr};
 
   try {
-    ScopedContext Active(hQueue->getContext());
+    ur_context_handle_t Ctx = hQueue->getContext();
+    ScopedContext Active(Ctx);
 
     uint32_t StreamToken;
     ur_stream_quard Guard;
     hipStream_t HIPStream = hQueue->getNextComputeStream(
         numEventsInWaitList, phEventWaitList, Guard, &StreamToken);
     hipFunction_t HIPFunc = hKernel->get();
 
+    hipDevice_t HIPDev = Ctx->getDevice()->get();
+    for (const void *P : hKernel->getPtrArgs()) {
+      auto [Addr, Size] = Ctx->getUSMMapping(P);
+      if (!Addr)
+        continue;
+      if (hipMemPrefetchAsync(Addr, Size, HIPDev, HIPStream) != hipSuccess)
```

Comment:
hipMemPrefetchAsync is kind of a "make resident" call, right?

Reply:
The docs are unclear, but from debugging the rocm/hsa driver, it seems that yes - memory won't be correctly tracked by the USM/HMM subsystem correctly until `hipMemPrefetchAsync` has been called. It's definitely a departure from the CUDA USM implementation that appears to automatically manage these allocations in all cases